jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -0,0 +1,277 @@
+   * following entries for the container file system:
+   *
+   * <ul>
+   *   <li>{@code /path/in/container/fileA}
+   *   <li>{@code /path/in/container/fileB}
+   *   <li>{@code /path/in/container/directory/}
+   *   <li>{@code /path/in/container/directory/...} (all contents of {@code directory/})
+   * </ul>
+   *
+   * @param files the source files to copy to a new layer in the container
+   * @param pathInContainer the path in the container file system corresponding to the {@code
+   *     sourceFile} (relative to root {@code /})
+   * @return this
+   * @throws IOException if an exception occurred when recursively listing any directories
+   */
+  public JibContainerBuilder addLayer(List<Path> files, Path pathInContainer) throws IOException {
+    LayerConfiguration.Builder layerConfigurationBuilder = LayerConfiguration.builder();
+
+    for (Path file : files) {
+      layerConfigurationBuilder.addEntryRecursive(
+          file, pathInContainer.resolve(file.getFileName()));
+    }
+
+    return addLayer(layerConfigurationBuilder.build());
+  }
+
+  /**
+   * Sets the layers (defined by a list of {@link LayerConfiguration}s). This replaces any
+   * previously-added layers.
+   *
+   * @param layerConfigurations the list of {@link LayerConfiguration}s
+   * @return this
+   */
+  public JibContainerBuilder setLayers(List<LayerConfiguration> layerConfigurations) {
+    this.layerConfigurations = new ArrayList<>(layerConfigurations);
+    return this;
+  }
+
+  /**
+   * Sets the layers. This replaces any previously-added layers.
+   *
+   * @param layerConfigurations the {@link LayerConfiguration}s
+   * @return this
+   */
+  public JibContainerBuilder setLayers(LayerConfiguration... layerConfigurations) {
+    return setLayers(Arrays.asList(layerConfigurations));
+  }
+
+  /**
+   * Adds a layer (defined by a {@link LayerConfiguration}).
+   *
+   * @param layerConfiguration the {@link LayerConfiguration}
+   * @return this
+   */
+  public JibContainerBuilder addLayer(LayerConfiguration layerConfiguration) {
+    layerConfigurations.add(layerConfiguration);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint. This is the beginning of the command that is run when the
+   * container starts. {@link #setProgramArguments} sets additional tokens.
+   *
+   * @param entrypoint a list of the entrypoint command
+   * @return this
+   */
+  public JibContainerBuilder setEntrypoint(List<String> entrypoint) {
+    this.entrypoint = ImmutableList.copyOf(entrypoint);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint.
+   *
+   * @param entrypoint the entrypoint command
+   * @return this
+   * @see #setEntrypoint(List) for more details
+   */
+  public JibContainerBuilder setEntrypoint(String... entrypoint) {
+    return setEntrypoint(Arrays.asList(entrypoint));
+  }
+
+  /**
+   * Sets the container entrypoint program arguments. These are additional tokens added to the end
+   * of the entrypoint command.
```

Comment:
I wonder if we should comment about how this corresponds to "Cmd" in the container config.

Reply:
Or maybe if this should just be cmd? Since that translates to both docker directly? But I guess kubernetes uses args?

Reply:
Yea, `cmd` is a confusing naming that I think Kubernetes tried to move away from.